relay-server/src/actors/processor.rs

```diff
@@ -2321,8 +2329,114 @@ mod tests {
 
     use crate::{actors::project::ProjectConfig, extractors::RequestMeta};
 
+    type ProcessSessionArguments = (
+        Item,
+        DateTime<Utc>,
+        Option<&'static str>,
+        Option<net::IpAddr>,
+        SessionMetricsConfig,
+        ClockDriftProcessor,
+        Vec<Metric>,
+    );
     use super::*;
 
+    // helper function that gives valid arguments to the process_session method
+    fn args_process_session() -> ProcessSessionArguments {
```

Comment:
This function a little bit confusing. And overall usage of tuple here as well. It might make sense create a proper struct for this and to be more explicit about the parameters used for test.

Reply:
do you think it's possible to write the testing better without refactoring the code? I'm not happy with it either but the function is not easy to unit test as I'm checking like 5 different conditions where it can return false early. 

The function is used only once yes, I tried to explain it with the comment above it. Perhaps it is better to make it a closure inside of the test function similar to `should_keep`.  

I see what you mean about making a struct instead, I could make that change.